docs/avro-data-source-guide.md

```diff
@@ -0,0 +1,267 @@
+---
+layout: global
+title: Avro Data Source Guide
```

Comment:
Call it "Apache Avro" in the title and first mention in the paragraph below. Afterwards, just "Avro" is OK.